src/screens/MainScreen/index.js

```diff
@@ -41,20 +48,39 @@ const MainScreen = () => {
     HIDDEN_VIEWS_POSITION,
   );
 
+  const onSelectTarget = useSelectTarget(
+    selectedTarget,
+    location,
+    coordsConstants,
+    mapView,
+    setSelectedTarget,
+    animate,
+    formPositionAnim,
+  );
+
   return (
     <View style={styles.container} testID={MAIN_SCREEN}>
       <MapView
+        ref={mapView}
         provider={PROVIDER_GOOGLE}
         style={styles.map}
-        region={{
+        initialRegion={{
           ...location,
-          latitudeDelta: LATITUDE_DELTA,
-          longitudeDelta: LONGITUDE_DELTA,
+          ...coordsConstants,
         }}
-        showUserLocation>
+        showUserLocation
+        followUserLocation>
         {topics &&
-          targets.map(({ target: { id, lat, lng, topicId } }) => (
-            <CustomMarker key={id} lat={lat} lng={lng} topicId={topicId} />
+          targets.map(({ target: { id, lat, lng, topicId } }, _, targets) => (
+            <TargetMarker
+              key={id}
+              id={id.toString()}
+              lat={lat}
+              lng={lng}
+              topicId={topicId}
+              onPress={() => onSelectTarget(id, targets)}
+              selected={selectedTarget && selectedTarget.id === id}
```

Comment:
Here you also can do:
```suggestion
              selected={selectedTarget?.id === id}
```